### ISSUE TITLE
ListObjects and writeMessage errors

### ISSUE THREAD

**Issue by jacobsa** [GoogleCloudPlatform/gcsfuse#101]
A couple of users (including @kernelt in #95) have reported errors that look like this:

```
fuse: 2015/08/04 07:55:31.358121 *fuseops.ReadDirOp error: readAllEntries: ReadEntries: ListObjects: EOF
fuse: 2015/08/04 07:55:31.358148 writeMessage: invalid argument
```

I can semi-reliably reproduce this now. It seems there are two bugs, that should be tackled in this order:

1. The error from `ReadDirOp` is apparently mishandled by `writeMessage`.
2. `ReadDirOp` gets the error in the first place.

**Comment by jacobsa** [comment]
It seems I can reproduce these errors with golang/go@a3ffd83, but not with golang/go@b3bf38e. I wonder if there was an HTTP bug fixed between those two commits...

**Comment by jacobsa** [comment]
Okay yeah, the `EOF` bug was introduced by golang/go@ 0cf48b4d919e5137ec03f1bf230fbd8720873151 and was fixed by golang/go@7e70c24.

I've confirmed by building with an old version of Go that commit fbda894c1bdf19bf3f3a959a01e4ec5032b61397 fixes the `writeMessage` issue. So all that remains is to make a new release built with a fixed version of Go.

**Comment by jacobsa** [comment]
Fixed with [v0.6.0](https://github.com/GoogleCloudPlatform/gcsfuse/releases/tag/v0.6.0).
